lib/ansible/modules/cloud/amazon/ec2_ami_copy.py

```diff
@@ -137,123 +132,85 @@
+              'Encrypted': module.params.get('encrypted'),
+#              'KmsKeyId': module.params.get('kms_key_id')
+              }
+    if module.params.get('kms_key_id'):
+        params['KmsKeyId'] = module.params.get('kms_key_id')
 
-    register_tags_if_any(module, ec2, tags, image_id)
-
-    module.exit_json(msg="AMI copy operation complete", image_id=image_id, state=img.state, changed=True)
-
-
-# register tags to the copied AMI
-def register_tags_if_any(module, ec2, tags, image_id):
-    if tags:
-        try:
-            ec2.create_tags([image_id], tags)
-        except Exception as e:
-            module.fail_json(msg=str(e))
-
-
-# wait here until the image is copied (i.e. the state becomes available
-def wait_until_image_is_copied(module, ec2, wait_timeout, img, image_id, wait):
-    wait_timeout = time.time() + wait_timeout
-    while wait and wait_timeout > time.time() and (img is None or img.state != 'available'):
-        img = ec2.get_image(image_id)
-        time.sleep(3)
-    if wait and wait_timeout <= time.time():
-        # waiting took too long
-        module.fail_json(msg="timed out waiting for image to be copied")
-    return img
-
-
-# wait until the image is recognized.
-def wait_until_image_is_recognized(module, ec2, wait_timeout, image_id, wait):
-    for i in range(wait_timeout):
-        try:
-            return ec2.get_image(image_id)
-        except boto.exception.EC2ResponseError as e:
-            # This exception we expect initially right after registering the copy with EC2 API
-            if 'InvalidAMIID.NotFound' in e.error_code and wait:
-                time.sleep(1)
-            else:
-                # On any other exception we should fail
-                module.fail_json(
-                    msg="Error while trying to find the new image. Using wait=yes and/or a longer wait_timeout may help: " + str(
-                        e))
-    else:
-        module.fail_json(msg="timed out waiting for image to be recognized")
+    try:
+        image_id = ec2.copy_image(**params)['ImageId']
+        if module.params.get('wait'):
+            ec2.get_waiter('image_available').wait(ImageIds=[image_id])
+        if module.params.get('tags'):
+            ec2.create_tags(
+                    Resources=[image_id],
+                    Tags=[{'Key' : k, 'Value': v} for k,v in module.params.get('tags').items()]
+                    )
+
+        module.exit_json(changed=True, image_id=image_id)
+    except ClientError as ce:
+        module.fail_json(msg=ce)
+    except NoCredentialsError:
+        module.fail_json(msg="Unable to locate AWS credentials")
+    except Exception as e:
+        module.fail_json(msg=str(e))
 
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
         source_region=dict(required=True),
         source_image_id=dict(required=True),
-        name=dict(),
-        description=dict(default=""),
+        name=dict(required=True),
```

Comment:
changing name to be required w/o updating docs, also see previous comment about changing existing options